src/components/Form.js

```diff
@@ -1,12 +1,64 @@
-const Form = () => (
-  <div>
-    <h2>ADD NEW BOOK</h2>
-    <form>
-      <input id="title" type="text" name="title" placeholder="Book title" />
-      <input id="author" type="text" name="author" placeholder="Author" />
-      <button type="button">ADD BOOK</button>
-    </form>
-  </div>
-);
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
+
+const Form = () => {
+  const dispatch = useDispatch();
+  const [newBook, setNewBook] = useState({
+    title: '',
+    author: '',
+    genre: '',
+  });
+
+  const handleInput = (e) => {
+    setNewBook({
+      ...newBook,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleClick = (e) => {
+    e.preventDefault();
+    const book = {
+      title: newBook.title,
+      author: newBook.author,
+      genre: newBook.genre,
+      id: uuidv4(),
+    };
+    dispatch(addBook(book));
+    setNewBook({
+      title: '',
+      author: '',
+      genre: '',
+    });
+  };
+
+  return (
+    <div>
+      <h2>ADD NEW BOOK</h2>
+      <form>
+        <input id="title" type="text" name="title" value={newBook.title} onInput={handleInput} placeholder="Book title" />
+        <input id="author" type="text" name="author" value={newBook.author} onInput={handleInput} placeholder="Author" />
+        <select name="genre" id="genre" value={newBook.genre} onInput={handleInput} placeholder="Genre">
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you make all your form `inputs` required to prevent users to submit an empty form. I suggest you add the `required` attribute in every input tag